### PR TITLE
Add unused GUPS parameter: number of updates/PE

### PR DESCRIPTION
**Pull request by davidozog**
Addressing #430.

Signed-off-by: David M. Ozog <david.m.ozog@intel.com>

### CODE REVIEW

File: test/apps/gups.c [hunk 1]
```diff
@@ -421,9 +421,13 @@ SHMEMRandomAccess(void)
...
 
   /* Default number of global updates to table: 4x number of table entries */
-  NumUpdates_Default = 4 * TableSize;
-  ProcNumUpdates = 4 * LocalTableSize;
-  NumUpdates = NumUpdates_Default;
+  if (NumUpdatesOpt == 0) {
+     ProcNumUpdates = 4 * LocalTableSize;
+     NumUpdates = 4 * TableSize;
+  } else {
+     ProcNumUpdates = NumUpdatesOpt;
```
Comment (jdinan) [inline test/apps/gups.c]: @davidozog Can you check against the MPI version of the benchmark to make sure this shouldn't be `ProcNumUpdates * NumUpdatesOpt`.

Comment (davidozog) [reply test/apps/gups.c]: [HPCC v1.5.0](http://icl.cs.utk.edu/hpcc/software/index.html) does not appear to support this parameter at all, it always performs `4 * TableSize` total updates.  

I'm fine just removing this adjustable parameter from the "Usage" output if you'd rather reflect HPCC.  But if we want it, either `ProcNumUpdates = NumUpdatesOpt` or `ProcNumUpdates = NumUpdatesOpt * LocalTableSize` would work, the former is just more fine-grained than the latter.  I'd guess the latter might be more puzzling to users though: "`-n`:  the scale factor of the local table size, which sets the total number of updates..." (or something similar) as opposed to: "`-n`:  the total number of updates."

Comment (jdinan) [reply test/apps/gups.c]: I filed #738 as a reminder to fix any remaining differences between our GUPS benchmark and HPCC RA.

Comment (jdinan) [reply test/apps/gups.c]: 🤷‍♂️ This seems reasonable to me.  At least the argument is not silently ignored any more.

Comment (jdinan) [reply test/apps/gups.c]: :shipit:

**Pull request merged**